Newsletters: add 136 (2021-02-17) (#517)
bitcoinops/bitcoinops.github.io: _posts/en/newsletters/2021-02-17-newsletter.md

  post] to the Bitcoin-Dev mailing list a draft BIP that describes how
  wallets, particularly hardware wallets, can securely exchange the
  information necessary to become signers for a multisig wallet.  The
  information that needs to be exchanged includes the script template to
  use (e.g.  P2WSH with 2-of-3 keys required to sign) and each signer's
  [BIP32][] extended public key (xpub) at the key path it plans to use
  for signing.

    Briefly, Nguyen's proposal (developed in correspondence with several
    hardware wallet producers) has a coordinator initiate the multisig
    federation process by choosing a script template and generating a
    private encryption and authentication credential.  Those parameters
    are shared with the member wallets, who choose an appropriate key
    path and return its xpub along with a signature generated by its
    corresponding private key.  The set of {identifying_parameters,
    key_path, xpub, signature} are encrypted and returned to the
    coordinator by each wallet.  The coordinator then combines them into
    an [output script descriptor][topic descriptors] that is encrypted

>     and passed back to each member wallet, who verifies their xpub is

Author: It's a symmetric cipher (AES), so everyone uses the same key (the key is derived from a short identifier that's key stretched to the required bits).

    included and then stores the descriptor as a script template they're
    willing to sign for.

    The proposal received a significant amount of discussion on list,
    and some changes are planned.  We'll monitor the discussion for
    significant updates as the proposal matures and gets closer to
    implementation.

- **Escrow over LN:** a discussion [started][zmn escrow1] on the
  Lightning-Dev mailing over a year ago about creating non-custodial
  onchain escrows for LN saw [renewed discussion][aragoneses escrow]
  this past week.  A particular standout in the discussion was a
  [post][zmn escrow2] by ZmnSCPxj that uses one of [De Morgan's laws][]

>   for boolean statements to greatly simply the construction of

Reviewer: simplify instead of simply?

  issue here has been [known][russell loop] for five years and discussed
  at length many times, so we expect discussion to continue.

- **Anti-exfiltration:** Andrew Poelstra and Jonas Nick published a blog
  [post][anti-exfil] about a security technique being implemented for
  the Shift [BitBox02][] and Blockstream [Jade][] hardware wallets.  The
  goal is to allow both a hardware wallet and the computer controlling
  it to each be assured that the nonce used to generate a signature is
  actually an unguessable value.  This prevents malicious hardware
  wallet firmware from generating nonces known to the firmware author,
  which the author could combine with your signatures found onchain to
  derive your private keys, allowing them to spend any other bitcoins
  controlled by those keys.  The post describes the technique used,
  which was previously mentioned in a mailing list thread about this
  subject almost a year ago (see Newsletters [#87][news87 exfil] and
  [#88][news88 exfil]).

## Changes to services and client software

*In this monthly feature, we highlight interesting updates to Bitcoin
wallets and services.*

- **Blockstream announces LNsync:**

>   [LNsync][blockstream blog lnsync] allows Lightning wallets to more quickly sync the

Reviewer 2: I believe "thin clients that need to catch up when intermittently connecting" is the main use case here. Although could be use more generally is my understanding.

  LN topology in order to optimally route a payment to its destination. An open
  source plugin, [historian][github historian], provides this functionality for
  C-Lightning users.

- **Rust light client Nakamoto released:**
  Alexis Sellier [released Nakamoto][nakamoto blog], a "high-assurance Bitcoin
  light-client implementation in Rust, with a focus on low resource utilization,

>   modularity and security".

Reviewer 2: Going to leave the quote as is since it is their words.

Good suggestion on noting the sync method (compact block filters). Added!

>   modularity and security".

Author: Although you're correct about the modification (and an additional reason to rewrite is the style rule that the items in a list with no other obvious ordering should appear in order of fewest to most syllables), I think this is too big of a rewrite for a quote.  Usually, I only modify punctuation/capitalization within quotes or use insertion/elision marks.

>   modularity and security".

Author: Also, the text says it uses "compact block filters", i.e. BIP157.

  satellite ln data].

- **Blockstream Green implements CSV:**
  Green Wallet has previously used nLockTime as a wallet recovery mechanism,
  requiring users to receive backup emails with pre-signed transactions from
  Blockstream after each transaction in order to recover funds. By [implementing
  an `OP_CHECKSEQUENCEVERIFY` (CSV) recovery mechanism][blockstream green csv],
  wallets can be recovered without needing backup files or associating an email
  address with the wallet.

- **Muun 2.0 released:**
  The latest [Muun 2.0 release][muun 2.0 blog] includes multisig, wallet recovery
  features, and also open sources the Android and iOS mobile apps.

- **Joinmarket 0.8.1 released:**
  The [0.8.1 release][joinmarket 0.8.1] contains additional [PSBT][topic psbt]
  support for PSBTs created externally, [signet][topic signet] support, a fix
  for BIP21 (see [Newsletter #127][news127 bech32 casing]). Terminal-based UI
  for JoinMarket, [JoininBox][github joininbox], was also updated to support 0.8.1.

- **VBTC adds batched segwit withdrawals:**
  VBTC, a Vietnamese exchange, [added batched segwit withdrawal][vbtc blog

>   segwit] capabilities after recently also [enabling Lightning withdrawals][vbtc blog lightning].

Reviewer 3: At first glance, the "Lightning Withdrawals" seem more noteworthy, so maybe that should be the title. Their process for batched withdrawals seems pretty involved, it's not clear to me what message we want to convey by highlighting it.

They charge a flat 1 mBTC fee for withdrawals and now offer this "Batched SegWit" withdrawal option. Withdrawing to a native segwit address gets an 80% discount, making the fee 0.2 mBTC. Other addresses "only" get discounted to 0.05 mBTC (almost certain that was supposed to be 0.5 mBTC). And the batched withdrawal is processed manually by the developers on the following weekend. The most interesting thing to me is that they actually offer financial encouragement to use the most efficient output format, but this is not highlighted in the brief summary.

I think this item should either be dropped or we should try to clarify what we think is noteworthy.

Reviewer 2: I think both the Lightning post (which is from a couple months ago which is why I had it take a back seat here a bit), and the segwit batching are newsworthy. There are some details around their batching process (as all batching has) and how they pass those savings on to their customers but I think adoption of batching and segwit (encouragement) is the main highlight.

I added some different wording here and a bit more detail. Hopefully that helps.

Reviewer 3: Looks much better, thanks!

---
title: 'Bitcoin Optech Newsletter #136'
permalink: /en/newsletters/2021/02/17/
name: 2021-02-17-newsletter
slug: 2021-02-17-newsletter
type: newsletter
layout: newsletter
lang: en
---
This week's newsletter mentions a new constant-time algorithm for
generating and verifying transaction signatures, describes a proposal to
stop processing unsolicited transactions, summarizes a proposed BIP for
setting up multisig wallets, shares insight from a discussion about
managing escrows over LN, links to renewed discussion about
bidirectional upfront LN fees, and mentions a new protocol for
mitigating the risk of malicious hardware wallets.  Also included are
our regular sections with news about updated clients and services,
announcements of new software releases and release candidates, and
descriptions of notable changes in popular Bitcoin infrastructure
software.

## News

- **Faster signature operations:** Russell O'Connor and Andrew Poelstra
  published a [blog post][safegcd bounds] announcing the implementation
  of an algorithm that may speed up Bitcoin Core's signature

>   verification by 15%.  It can also reduce by 25% the amount of time it

Reviewer 4: "may speed up" and "can reduce" indicate different levels of certainty to me, perhaps "speeds up" and "reduces", or "may speed up" and "may reduce"

  before receiving payment.  The operation requires an upgrade of LN to
  use [PTLCs][topic ptlc], which is not currently expected until after
  [taproot][topic taproot] activates.

- **Renewed discussion about bidirectional upfront LN fees:**  Joost
  Jager [restarted][jager hold fees] Lightning-Dev mailing list
  discussion about adding LN service fees that charge spenders and
  receivers for the time they consume using a channel's limited
  availability of funding ("liquidity") and concurrent payment capacity
  ("HTLC slots").  Jager builds on a previous proposal (see [Newsletter
  #122][news122 bidir]), extending its fixed fees with fees
  proportional to payment processing duration ("hold fees").  The
  proposal received moderate discussion, with one [concern][zmn hold fee
  privacy] being a reduction in sender/receiver privacy.  The underlying
  issue here has been [known][russell loop] for five years and discussed
  at length many times, so we expect discussion to continue.

- **Anti-exfiltration:** Andrew Poelstra and Jonas Nick published a blog
  [post][anti-exfil] about a security technique being implemented for

>   the Shift [BitBox02][] and Blockstream [Jade][] hardware wallets.  The

Reviewer 4: Shift Crypto 
```suggestion
  the Shift Crypto [BitBox02][] and Blockstream [Jade][] hardware wallets.  The
```

  goal is to allow both a hardware wallet and the computer controlling
  it to each be assured that the nonce used to generate a signature is
  actually an unguessable value.  This prevents malicious hardware
  wallet firmware from generating nonces known to the firmware author,
  which the author could combine with your signatures found onchain to

>   derive your private keys, allowing them to spend any other bitcoins

Reviewer 4: the shift from 3rd to 2nd person in this sentence (using "your" twice instead of user/client/node/etc.) is a bit surprising; it's not used elsewhere in this newsletter

  [#88][news88 exfil]).

## Changes to services and client software

*In this monthly feature, we highlight interesting updates to Bitcoin
wallets and services.*

- **Blockstream announces LNsync:**
  [LNsync][blockstream blog lnsync] allows Lightning wallets to more quickly sync the
  LN topology in order to optimally route a payment to its destination. An open
  source plugin, [historian][github historian], provides this functionality for
  C-Lightning users.

- **Rust light client Nakamoto released:**
  Alexis Sellier released [Nakamoto][nakamoto blog], a "high-assurance Bitcoin
  light-client implementation in Rust, with a focus on low resource utilization,
  modularity and security" based on [compact block filters][topic compact block filters].

- **Blockstream Satellite broadcasting LN data and Bitcoin Core source:**
  Blockstream's Satellite, in addition to broadcasting Bitcoin block chain data,
  now includes the [Bitcoin Core source code][blockstream satellite bitcoin
  code], code for a satellite-optimized Bitcoin fork ([Bitcoin
  Satellite][github bitcoin satellite]), and [LN gossip data][blockstream
  satellite ln data].

- **Blockstream Green implements CSV:**

>   Green Wallet has previously used nLockTime as a wallet recovery mechanism,

Reviewer 4: ```suggestion
  Green Wallet previously used nLockTime as a wallet recovery mechanism,
```

  requiring users to receive backup emails with pre-signed transactions from
  Blockstream after each transaction in order to recover funds. By [implementing
  an `OP_CHECKSEQUENCEVERIFY` (CSV) recovery mechanism][blockstream green csv],

>   wallets can be recovered without needing transaction-specific backup files or

Reviewer 4: ```suggestion
  wallets can now be recovered without needing transaction-specific backup files or
```

  associating an email address with the wallet.

- **Muun 2.0 released:**
  The [2.0 release][muun 2.0 blog] of mobile Bitcoin and Lightning wallet Muun
  includes multisig, wallet recovery features, and also open sources the Android
  and iOS mobile apps.

- **Joinmarket 0.8.1 released:**
  The [0.8.1 release][joinmarket 0.8.1] contains additional [PSBT][topic psbt]

>   support for PSBTs created externally, [signet][topic signet] support, a fix

Reviewer 4: ```suggestion
  support for PSBTs created externally, [signet][topic signet] support, and a fix
```

>   for BIP21 (see [Newsletter #127][news127 bech32 casing]). A terminal-based UI

Reviewer 4: perhaps a couple more words about the BIP21 fix

  for JoinMarket, [JoininBox][github joininbox], was also updated to support 0.8.1.

- **VBTC adds batched segwit withdrawals:**
  VBTC, a Vietnamese exchange, [added a batched segwit withdrawal option][vbtc blog
  segwit] after recently also [enabling Lightning withdrawals][vbtc blog
  lightning]. The cheaper segwit batch withdrawal occurs once a week,

>   targeting a time when a mempool transaction backlog is likely to be low.

Reviewer 4: ```suggestion
  targeting a time when the mempool transaction backlog is likely to be low.
```

- [Rust-Lightning #787][] ensures that channel closures due to error messages
  only happen when the peer who sent the message is the channel's counterparty.
  Previously, it was possible for malicious peers to force-close any channel if
  they knew the channel id.

- [BTCPay Server #2164][] redesigns the wallet setup wizard, walking the user
  through setting up BTCPay's internal wallet and optionally integrating
  it with the user's own remote software or hardware wallet.  This is
  the start of other redesign work on BTCPay's interface.

- [HWI #443][] adds support for signing multisig inputs with the
  BitBox02 hardware wallet.

- [Bitcoin Core #19145][] extends the `gettxoutsetinfo` RPC's
  `hash_type` option to accept a new `muhash` parameter that will
  produce the [MuHash3072][] digest of the UTXO set at the current block
  height.  This is an alternative to producing the default SHA256 digest of
  the UTXO set.  When run this way, MuHash has to process the same
  amount as data as the SHA256 function, so there's not expected to be
  any significant change in performance of the `gettxoutsetinfo` RPC,
  which can take up to several minutes to return on slow single-board

>   computers.  However, a previously-calculated MuHash digest can have

Reviewer 5: I call the finalized `uint256` result of the MuHash object the 'digest' and the intermittent `MuHash3072` object 'muhash object'. The object is what can be updated quickly.

```suggestion
  computers.  However, a previously-calculated MuHash object can have
```

Author: Oh, that's a good distinction; I feel dumb now for treating those two obviously different things as the same.  Thanks!

>   computers.  However, a previously-calculated MuHash digest can have

Reviewer 4: nit
```suggestion
  computers.  However, a previously calculated MuHash digest can have
```

  elements added or removed from it relatively cheaply, so a future PR

>   is expected to quickly calculate a MuHash digest of the UTXO set at
>   each block, save those results in a simple database, and allow
>   returning them almost instantly upon demand.  This will also benefit

Reviewer 5: ```suggestion
  is expected to add a rolling MuHash object representing the UTXO set at
  each block height, save finalized digests for each height in an index, and allow
  returning them almost instantly upon demand.  This will also benefit
```

>   the [AssumeUTXO][topic assumeutxo] project, which will rely on the

Reviewer 4: ```suggestion
  the [AssumeUTXO][topic assumeutxo] project, which relies on the
```

  ability of users to compare UTXO set hashes at selected past block
  heights.

- [C-Lightning #4364][] changes how it notifies its channel partners
  about problems, separating existing error messages into true *errors*
  and those that are just *warnings*.  The current [BOLT1][]

>   specification requires any error result in channels being closed

Reviewer 4: "requires that any error result" or "requires any error to result"

  (although this does not seem to be universally implemented), but a
  [proposed specification update][bolts #834] introduces the warning
  message type that can allow for a more flexible response.  Anyone
  interested may also wish to read a [post][kirk-cohen post] to the
  Lightning-Dev mailing list this week by Carla Kirk-Cohen about

>   extended error message descriptions, a topic she described as,

Reviewer 4: ```suggestion
  extended error message descriptions, a topic she described as
```

*In this monthly feature, we highlight interesting updates to Bitcoin
wallets and services.*

- **Blockstream announces LNsync:**
  [LNsync][blockstream blog lnsync] allows Lightning wallets to more quickly sync the
  LN topology in order to optimally route a payment to its destination. An open
  source plugin, [historian][github historian], provides this functionality for
  C-Lightning users.

- **Rust light client Nakamoto released:**
  Alexis Sellier [released Nakamoto][nakamoto blog], a "high-assurance Bitcoin
  light-client implementation in Rust, with a focus on low resource utilization,
  modularity and security".

> - **Blockstream Satellite broadcasting LN data, Bitcoin Core source:**

Author: Nit: in general I think it's better to conjoin two items with a traditional conjunction such as "and" rather than a comma.  However, I'm not opposed to the above.

>   Blockstream's Satellite, in addition to broadcasting Bitcoin blockchain data,

Author: Nit: style guide suggests "block chain" as two words: https://github.com/bitcoinops/bitcoinops.github.io/blob/master/STYLE.md#compound-words

  now includes the [Bitcoin Core source code][blockstream satellite bitcoin
  code], code for a satellite-optimized Bitcoin fork, [Bitcoin

>   Satellite][github bitcoin satellite], and [LN gossip data][blockstream

Author: When embedding a comma-separated parenthetical ("Bitcoin Satellite") inside a list, the usual rule is to separate items in the list with semicolons rather than commas (or to put the parenthetical inside parenthesis, or to rewrite so that none of these solutions is needed).

Reviewer 2: Thanks for the rec, see if the latest push properly addresses this.

  satellite ln data].

- **Blockstream Green implements CSV:**
  Green Wallet has previously used nLockTime as a wallet recovery mechanism,
  requiring users to receive backup emails with pre-signed transactions from
  Blockstream after each transaction in order to recover funds. By [implementing
  an `OP_CHECKSEQUENCEVERIFY` (CSV) recovery mechanism][blockstream green csv],

>   wallets can be recovered without needing backup files or associating an email

Author: Maybe: s/backup files/special backup files/.  That may make it clearer that whatever usual backup stuff Green requires (e.g. an HD seed) is still needed.

  address with the wallet.

> - **Muun 2.0 released:**

Author: Maybe mention what Muun is; I'd never heard of it before.

  The latest [Muun 2.0 release][muun 2.0 blog] includes multisig, wallet recovery
  features, and also open sources the Android and iOS mobile apps.

- **Joinmarket 0.8.1 released:**
  The [0.8.1 release][joinmarket 0.8.1] contains additional [PSBT][topic psbt]
  support for PSBTs created externally, [signet][topic signet] support, a fix

>   for BIP21 (see [Newsletter #127][news127 bech32 casing]). Terminal-based UI

Author: I think this sentence should probably start with an article: either "A terminal-based [...]" or "The terminal-based [...]".

  for JoinMarket, [JoininBox][github joininbox], was also updated to support 0.8.1.

- **VBTC adds batched segwit withdrawals:**
  VBTC, a Vietnamese exchange, [added batched segwit withdrawal][vbtc blog
  segwit] capabilities after recently also [enabling Lightning withdrawals][vbtc blog lightning].

- **Bitcoin Dev Kit v0.3.0 released:**
  Rust wallet library Bitcoin Dev Kit announced its [v0.3.0 release][bdk
  0.3.0 blog] including splitting out the CLI into its own repository. The
  recent [BDK v0.2.0][bdk 0.2.0 blog] release added Branch and Bound (BnB) coin selection,

>   [descriptor][topic descriptors] templates, `sortedmulti`, and more.

Author: Since `sortedmulti` is part of descriptors, maybe make it a parenthentical: "[...], descriptor templates (including with the recently-added `sortedmulti`), and more"
